src/diffusers/textual_inversion_utils.py

```diff
@@ -0,0 +1,101 @@
+import torch
+
+
+class TextualInversionMixin:
+    r"""
+    Mixin class for adding textual inversion tokens and embeddings to the tokenizer and text encoder with method:
+    - [`~TextualInversionMixin.load_textual_inversion_embeddings`]
+    - [`~TextualInversionMixin.add_textual_inversion_embedding`]
+
+    Class attributes:
+    - **textual_inversion_tokens** (`List[str]`): list of tokens added to the tokenizer's vocabulary and the text
+      encoder's embedding matrix
+    """
+    textual_inversion_tokens = []
+
+    def load_textual_inversion_embeddings(self, embeddings):
+        r"""
+        Loads textual inversion embeddings.
+
+        Receives a dictionary with the following keys:
+        - `token`: name of the token to be added to the tokenizers' vocabulary
+        - `embedding`: path to the embedding of the token to be added to the text encoder's embedding matrix
+
+        Alternatively, it can receive a list of pathes to embedding dictionaries, where the keys are the tokens and the
+        values are the embeddings. In that case, it will iterate through the list and add the tokens and embeddings to
+        the tokenizer's vocabulary and the text encoder's embedding matrix.
+
+        Iters through the dictionary and adds the token to the tokenizer's vocabulary and the embedding to the text
+        encoder's embedding matrix.
+        """
+
+        if isinstance(embeddings, dict):
+            for token, embedding_path in embeddings.items():
+                # check if token in tokenizer vocab
+                # if yes, raise exception
+                if token in self.tokenizer.get_vocab():
```

Comment:
What about an option `allow_replace` that allows one to redefine existing tokens? This would also allow to add changed embeddings without reloading the vanilla model only to be able to replace on embedding by a newer version.

Reply:
Great suggestion. I implemented this argument and tested in production already -- works great for dynamically updating and testing different checkpoints for the same token on the fly.